tests/agent_server/deploy/e2e/test_autostarted.py

```diff
@@ -244,3 +252,677 @@ def repaired() -> bool:
+                postgres_processes.append(process)
+            case _:
+                other_processes.append(process)
+
+    return {
+        "inmanta fork server": inmanta_fork_server,
+        "inmanta executor": inmanta_executor,
+        "python": python_processes,
+        "pg_ctl": postgres_processes,
+        "other": other_processes,
+    }
+
+
+def wait_for_terminated_status(current_children: list[psutil.Process], expected_terminated_process: int = 1) -> bool:
+    """
+    Check that the number of terminated processes matches the expected ones
+
+    :param current_children: Current processes
+    :param expected_terminated_process: How many processes should be "terminated"
+    """
+    terminated_process = []
+    for process in current_children:
+        try:
+            if process.status() == "terminated":
+                terminated_process.append(process)
+        except psutil.NoSuchProcess:
+            terminated_process.append(None)
+
+    return len(terminated_process) == expected_terminated_process
+
+
+@pytest.fixture
+async def ensure_consistent_starting_point(agent: Agent, ensure_resource_tracker_is_started: None) -> int:
+    """
+    Make sure that every test that uses this fixture will begin in a consistent, i.e.:
+        - 2 processes: a postgres server and the scheduler
+        - 3 processes:
+            - a postgres server, the scheduler and the resource tracker
+            - a postgres server, the scheduler and one inmanta fork server (unrelated to the new scheduler)
+        - 4 processes: a postgres server, the scheduler, one inmanta fork server and the resource tracker
+
+    :param agent: The agent fixture (that we want to stop before running the test)
+    :param ensure_resource_tracker_is_started: The fixture that creates a Resource Tracker process
+    """
+
+    def is_consistent_state(current_processes: list[psutil.Process], process_mapping: dict[str, list[psutil.Process]]) -> bool:
+        """
+        Returns True if we are in one of the above situation, False otherwise
+
+        :param current_processes: Current processes
+        :param process_mapping: Current mapping of processes
+        """
+        match len(current_processes):
+            case 2:
+                return len(process_mapping["pg_ctl"]) == 1 and len(process_mapping["python"]) == 1
+            case 3:
+                return (
+                    len(process_mapping["pg_ctl"]) == 1
+                    and len(process_mapping["python"]) == 2
+                    or len(process_mapping["pg_ctl"]) == 1
+                    and len(process_mapping["python"]) == 1
+                    and len(process_mapping["inmanta fork server"]) == 1
+                )
+            case 4:
+                return (
+                    len(process_mapping["pg_ctl"]) == 1
+                    and len(process_mapping["python"]) == 2
+                    and len(process_mapping["inmanta fork server"]) == 1
+                )
```

Comment:
Why this odd structure? Why not just pass in a list of processes and remove these that are allowed in stead of enumerating the possibilities?

Is it sufficient to check only `python` and nothing more?